### ISSUE TITLE
Unable to replicate the results

### ISSUE THREAD

**Issue by Mayurji** [rymc/n2d#15]
I am not sure this question comes under issues or not ? But I'm still gonna ask.

Hi @rymc, I am trying to replicate the results on MNIST dataset alone. I am little confused with the way the model is getting trained.

* creating an autoencoder model #line354
* extracting only modules related to encoder alone, creating separate model for encoder #line357.
* then training the autoencoder model.
* But predicting on Images, using encoder model ? #line385

How the encoder model got trained, since we are using it to predict on images?
Thanks in advance.

**Comment by rymc** [comment]
Hi,

Thank you for your interest. 

The creation of a separate model on line 357 is done via an extraction of a reference to the encoder of the autoencoder. When the autoencoder is later trained, the encoder model is using the trained representation from the encoder of the autoencoder.

The prediction on line 385 is the process of encoding the input image using the autoencoder we trained. Perhaps the function name 'predict' is a little misleading as it is not predicting the class of the data, rather it is encoding the data. 

As this is an unsupervised method, there isn't really any prediction step (in the classification sense) at all. Instead, we are clustering the images. We use the labels of the images only to measure the quality of the resulting clusters.

If you've any questions about the ideas behind this, feel free to send me an email. 

If you've any issues reproducing the results, feel free to open a new issue.

**Comment by Mayurji** [comment]
Thanks @rymc, I have managed to replicate the results using autoencoder implementation in Pytorch.

**Comment by abduallahmohamed** [comment]
@Mayurji are you able to share your Pytorch implementation?

**Comment by Mayurji** [comment]
@abduallahmohamed, Yes I am able to replicate the results on MNIST and for other dataset, we can create pytorch dataloader. Check the github repo for model implementation.
